sql/core/src/test/scala/org/apache/spark/sql/execution/streaming/state/ListStateSuite.scala

```diff
@@ -99,6 +99,153 @@ class ListStateSuite extends StateVariableSuiteBase {
     }
   }
 
+  // verify that relative ordering of inserted elements is maintained on retrieval - long type
```

Comment:
The expected behavior is the same with Array/List in Scala, right? Let's retain the expectation and apply the same operation, and compare. It'd be lot easier than enumerating all constants across multiple lines.

I think this would help simplifying the test code a lot, especially for case class.

Reply:
Done